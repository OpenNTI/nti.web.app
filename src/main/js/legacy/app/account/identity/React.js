const Ext = require('@nti/extjs');
const { getService } = require('@nti/web-client');
const { Identity } = require('@nti/web-profiles');
const LoginActions = require('internal/legacy/login/Actions');
require('internal/legacy/overrides/ReactHarness');

const AccountActions = require('../Actions');

module.exports = exports = Ext.define(
	'NextThought.app.account.identity.React',
	{
		extend: 'NextThought.ReactHarness',
		alias: 'widget.identity-react',
		component: Identity.Session,

		setTheme(scope, knockout) {},

		initComponent: function () {
			this.callParent(arguments);
			Object.assign(this.initialConfig, {
				addHistory: true,
				baseroute: '/app',
			});
			this.AccountActions = AccountActions.create();
			this.LoginActions = LoginActions.create();
		},

		async afterRender() {
			this.callParent(arguments);
			const service = await getService();
			const supportLinks = service.getSupportLinks();

			this.mon(Ext.getBody(), {
				click: this.onBodyClicks.bind(this, supportLinks),
			});
		},

		onBodyClicks(supportLinks, e) {
			let action = e.getTarget('a[data-action]')?.dataset.action;
			const link = e.getTarget('a[href]');
			const href = link?.getAttribute('href');

			// This is used to launch support logic the same way everywhere
			// we have a link that goes to the supportContact url. (so we can
			// just emit that url in the dom and not worry about coping logic)
			if (
				supportLinks.supportContact === href &&
				supportLinks.internalSupport
			) {
				action = action || 'support';
			}

			if (action) {
				if (call(this, `do-${action}`, link, e) !== false) {
					e.stopEvent();
				}
			}
		},

		showLink(target) {
			target.setAttribute('target', '_blank');
			return false;
		},

		'do-welcomeGuide'(target) {
			this.AccountActions.showWelcomePage(target.href);
		},

		'do-about': 'showLink',

		'do-privacy'(target) {
			this.AccountActions.showPrivacy(target);
		},

		'do-privacyForMinors'(target) {
			this.AccountActions.showChildrensPrivacy(target.href);
		},

		'do-terms'(target) {
			this.AccountActions.showTermsOfService(target.href);
		},

		'do-support'(target) {
			this.AccountActions.showContactUs();
		},

		'do-helpSite': 'showLink',

		'do-impersonate'() {
			this.LoginActions.handleImpersonate();
		},

		'do-logout'() {
			this.LoginActions.handleLogout();
		},
	}
);

function call(scope, action, ...args) {
	if (typeof action === 'string') {
		return call(scope, scope[action], ...args);
	}

	return action.apply(scope, args);
}
