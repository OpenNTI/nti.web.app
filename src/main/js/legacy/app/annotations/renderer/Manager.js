var Ext = require('extjs');
var Anchors = require('../../../util/Anchors');
var Globals = require('../../../util/Globals');
var UtilLine = require('../../../util/Line');


module.exports = exports = Ext.define('NextThought.app.annotations.renderer.Manager', {
    events: new Ext.util.Observable(),
    registry: [],

    constructor: function(reader) {
		this.callParent();
		this.reader = reader;
		this.rendererSuspended = 0;
	},

    controlLineTmpl: Ext.DomHelper.createTemplate({ cls: 'controlContainer'}).compile(),
    isDebug: false,

    /**
	 * @constructor Inner class
	 */
	Bucket: function() {
		this.values = {};
		this.length = 0;

		this.free = function() {
			this.each(function(v, k, o) { delete o[k]; }, this);
			Ext.Object.each(this, function(k, o, me) { delete me[k]; });
		};

		this.put = function(o, key) {
			key = typeof key === 'number' ? key : guidGenerator();
			if (this.values[key]) {
				throw 'existing value';
			}
			this.values[key] = o;
			this.length += 1;
			return key;
		};

		this.each = function(cb, s) {
			Ext.Object.each(this.values, function(key, value, o) {
				Ext.callback(cb, s, [value, key, o]);
			});
		};

		this.get = function(key) {
			return this.values[key];
		};

		this.first = function() {
			var key;
			for (key in this.values) {
				if (this.values.hasOwnProperty(key)) {
					return this.get(key);
				}
			}
			return null;
		};
	},

    registerGutter: function(el, reader) {
		//TODO all this junk about prefixes should go away once we aren't using a singleton here...
		if (this.gutter) {
			console.warn('replacing exisiting gutter?', this.gutter);
		}
		this.gutter = el;

		//el.addCls('debug');

		el.controls = el.down('.controls');

		this.render();//renders wait for the gutter to exist
	},

    register: function(o) {
		this.registry.push(o);
		o.requestRender();
	},

    unregister: function(o) {
		var r = this.registry;
		if (r) {
			this.registry = Ext.Array.remove(r, o);
			if (this.registry.length === 0) {
				this.sorter = null;
			}
		}
	},

    getReader: function() {
		return this.reader;
	},

    buildSorter: function(prefix) {
		//Default sort will sort by lastModified
		return function(a, b) {
			var e = {get: Ext.emptyFn},
					c = 0, $a = (a.record || e).get('Last Modified'), $b = (b.record || e).get('Last Modified');
			if ($a !== $b) {
				c = $a < $b ? 1 : -1;
			}

			return c;
		};
	},

    clearBuckets: function() {
		function clear(d) {
			while (d && d.firstChild) {
				d.removeChild(d.firstChild);
			}
		}

		var g = this.gutter;

		if (this.gutter.down('.active')) {
			this.selectedLine = this.gutter.down('.active').getAttribute('data-line');
		}
		clear(g.controls.dom);

		if (this.buckets) {
			this.buckets.free();
		}
		this.buckets = new this.Bucket();
	},

    getBucket: function(line) {
		//console.debug('prefix:'+prefix, line);
		if (line < 0) {
			//bad line, don't render:
			if (this.isDebug) {
				console.error('Annotation cannot be rendered in gutter');
			}
			return null;
		}

		var c = this.buckets,
			lineTolerance = 40,
			b = c ? c.get(line) : null,
			a = Math.round(line - (lineTolerance / 2)), z = Math.round(line + (lineTolerance / 2));

		if (!b && c) {
			c.each(function(value, key) {
				var keyNum = parseInt(key, 10);
				if (keyNum >= a && keyNum <= z) {
					b = value;
					return false;
				}
				return true;
			});

			if (!b) {
				b = new this.Bucket();
				b.line = line;
				c.put(b, line);
			}
		}
		return b;
	},

    layoutBuckets: function() {
		var g = this.gutter,
			b = this.buckets,
			cT = this.controlLineTmpl;


		b.each(function(line, y) {
			var count = 0;

			line.controls = line.controls || cT.append(g.controls, [], true);
			line.controls.setStyle('top', y + 'px');
			line.controls.set({'data-line': y});

			line.each(function(o) {
				if (o.isNote) {
					++count;
				}
			});
			if (count > 0) {
				Ext.DomHelper.overwrite(line.controls, {cls: 'count', html: count});
			}

			if (!line.controls.first()) {
				line.controls.remove();
				delete line.controls;
			}
		});
	},

    suspend: function() {
		this.rendererSuspended++;
	},

    resume: function() {
		this.rendererSuspended--;
		if (this.rendererSuspended === 0 && this.renderOnResume) {
			delete this.renderOnResume;
			this.render();
		}
	},

    render: function() {
		var me = this, containers = {}, renderedCount = 0,
			cleanContent, rootContainerId,
			cloned, descs = [], cids = [], doc = null, selectedEl;

		if (me.rendering) {
			console.warn('Render called while rendering...');
			me.events.on('finish', Ext.bind(me.render, me, arguments), me, {single: true});
			return;
		}
		if (this.rendererSuspended > 0) {
			this.renderOnResume = true;
			return;
		}
		if (!me.gutter) {
			console.error('no gutter');
			me.events.fireEvent('rendering');
			me.events.fireEvent('finish', 0);
			return;
		}

		if (!me.registry) {
			me.events.fireEvent('rendering');
			me.events.fireEvent('finish', 0);
			return;//nothing to do
		}

		me.rendering = true;
		me.events.fireEvent('rendering');
		console.log('Rendering annotations');
		if (console.time) {
			console.time('Annotation render loop');
		}
		Ext.suspendLayouts();

		me.registry = Ext.Array.unique(me.registry);

		me.sorter = me.sorter || me.buildSorter();
		if (me.sorter) {
			Ext.Array.sort(me.registry, me.sorter);
		}

		me.clearBuckets();

		cloned = Ext.Array.clone(me.registry);
		Ext.each(cloned, function(o) {
			var desc = o.getRecordField ? o.getRecordField('applicableRange') : null,
					cid = o.getRecordField ? o.getRecordField('ContainerId') : null;
			if (o.doc) {
				doc = o.doc;
			}
			descs.push(desc);
			cids.push(cid);
		});

		if (cloned && cloned.length > 0 && doc) {
			cleanContent = me.getReader().getCleanContent();
			rootContainerId = Anchors.rootContainerIdFromDocument(doc);
			Anchors.preresolveLocatorInfo(descs, doc, cleanContent, cids, rootContainerId);
			Ext.each(cloned, function(o) {
				var y, b, c;
				try {
					if (!o.isVisible) {
						return;
					}

					c = containers[o.getContainerId()] = (containers[o.getContainerId()] || []);

					if (o.isNote) {
						c.push(o);
					}

					y = o.render();
					if (y !== NextThought.app.annotations.Base.HIDDEN) {
						b = me.getBucket(Math.ceil(y));
						if (b) {
							o.getRecord().set('line', b.line);
							renderedCount++;
							b.put(o);
						}
					}
				}
				catch (e) {
					console.error(o.$className, Globals.getError(e));
				}
			});
			console[renderedCount === cloned.length ? 'log' : 'warn']('Rendered ' + renderedCount + '/' + cloned.length + ' annotations');
			me.layoutBuckets();
		}

		me.rendering = false;
		Ext.resumeLayouts(true);
		me.events.fireEvent('finish', renderedCount);
		if (console.timeEnd) {
			console.timeEnd('Annotation render loop');
		}
		if (this.selectedLine) {
			selectedEl = this.gutter.down('[data-line=' + this.selectedLine + ']');
			if (selectedEl) {
				selectedEl.addCls('active');
			}
		}
	}
}, function() {
	var me = this,
			fn = this.prototype.render,
			timerId;

	me.prototype.render = function() {
		var callerScope = this;
		if (timerId) {
			clearTimeout(timerId);
		}
		timerId = setTimeout(function() {
			fn.call(callerScope);
		}, 100);
	};
});
