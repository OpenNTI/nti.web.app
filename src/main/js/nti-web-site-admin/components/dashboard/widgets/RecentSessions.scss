@import "~nti-style-common/all.scss";

.recent-sessions {
	width: 504px;
	height: 369px;

	.title {
		margin-top: 10px;
	}

	.items-container {
		display: flex;

		.overall-stats {
			margin: 0 auto;
		}

		.items {
			width: 100%;

			.item {
				.info {
					padding-top: 0;

					.main-info {
						display: flex;

						.duration {
							margin: 2px 0 0 16px;
						}
					}
				}

				.item-image {
					border-radius: 30px;
					min-width: 40px;
				}

				.start-time, .duration {
					font-size: 12px;
					color: #b8b8b8;
				}

				.value {
					right: 16px;

					.app-icon {
						.screen {
							width: 22px;
							height: 18px;
							border-radius: 4px;
							border: solid 2px #ccc;
							text-align: center;
						}

						.base {
							width: 22px;
							height: 8px;
							background-color: #ccc;
							border-radius: 2px;
							text-align: center;
						}
					}

					.mobile-icon {
						margin-right: 2px;

						.top {
							width: 18px;
							height: 4px;
							border-radius: 4px 4px 0 0;
							background-color: #ccc;
							text-align: center;
						}

						.screen {
							width: 18px;
							height: 24px;
							border: solid 1px #ccc;
							text-align: center;
						}

						.base {
							width: 18px;
							height: 6px;
							background-color: #ccc;
							border-radius: 0 0 4px 4px;
							text-align: center;
						}
					}
				}
			}
		}
	}
}
