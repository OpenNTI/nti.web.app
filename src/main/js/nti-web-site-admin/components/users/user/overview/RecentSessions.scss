@import "~@nti/style-common";

.user-recent-sessions {
	width: 241px;
	height: 340px;
	box-shadow: 0 1px 2px 1px #ccc;
	color: #555;
	font-weight: 300;
	padding: 10px;

	.header {
		.title {
			margin-left: 10px;
			font-size: 24px;
			font-weight: 500;
		}
	}

	figure.loading {
		position: relative;
		top: 20%;
	}

	.no-sessions {
		margin-left: 12px;
		margin-top: 20%;
	}

	.items {
		margin: 10px;

		.item {
			margin: 10px 0;

			.day {
				font-size: 12px;
				font-weight: 500;
				color: #b8b8b8;
			}
		}
	}
}
