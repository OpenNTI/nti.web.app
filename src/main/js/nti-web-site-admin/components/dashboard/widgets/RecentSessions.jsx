import './RecentSessions.scss';
import React from 'react';
import PropTypes from 'prop-types';

import { scoped } from '@nti/lib-locale';
import { DateTime, Loading, Avatar } from '@nti/web-commons';
import { getService, User } from '@nti/web-client';
import { getLink } from '@nti/lib-interfaces';

const ANALYTICS = 'Analytics';
const SESSIONS_LINK = 'sessions';
const LIMIT = 10;
const PAGE_SIZE = 6;

const LABELS = {
	title: 'Recent Sessions',
	name: 'Name',
	value: '',
	noItems: 'No sessions found',
};

const t = scoped(
	'nti-web-site-admins.components.dashboard.widgets.recentsessions',
	LABELS
);

class Item extends React.Component {
	static propTypes = {
		item: PropTypes.object.isRequired,
	};

	renderImg() {
		const { item } = this.props;

		return <Avatar className="item-image" entity={item.user} />;
	}

	getDurationString(duration) {
		const hours = Math.floor(duration / (60 * 60));
		const minutes = Math.floor((duration - hours * 60 * 60) / 60);
		const seconds = duration - hours * 60 * 60 - minutes * 60;

		return (
			(hours > 0 ? hours + 'h ' : '') +
			(minutes > 0 || hours > 0 ? minutes + 'm ' : '') +
			(seconds + 's')
		);
	}

	renderInfo() {
		const { item } = this.props;

		const duration = item.SessionEndTime
			? this.getDurationString(
					item.SessionEndTime - item.SessionStartTime
			  )
			: 'Ongoing';

		return (
			<div className="info">
				<div className="main-info">
					<div className="name">{item.user.alias}</div>
					<div className="duration">{duration}</div>
				</div>
				<div className="start-time">
					{DateTime.fromNow(item.SessionStartTime * 1000)}
				</div>
			</div>
		);
	}

	renderMobileIcon() {
		return (
			<div className="mobile-icon">
				<div className="top" />
				<div className="screen" />
				<div className="base" />
			</div>
		);
	}

	renderAppIcon() {
		return (
			<div className="app-icon">
				<div className="screen" />
				<div className="base" />
			</div>
		);
	}

	renderValue() {
		const { item } = this.props;

		return (
			<div className="value">
				{item.appType === 'mobile'
					? this.renderMobileIcon()
					: this.renderAppIcon()}
			</div>
		);
	}

	render() {
		return (
			<div className="item">
				{this.renderImg()}
				{this.renderInfo()}
				{this.renderValue()}
			</div>
		);
	}
}

export default class RecentSessions extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			loading: true,
			pageNumber: 0,
		};
	}

	componentDidMount() {
		this.setState({ items: [] }, () => {
			this.loadData();
		});
	}

	resolveUser(userName) {
		return User.resolve({ entity: userName })
			.then(user => user)
			.catch(() => {
				return {
					alias: userName,
				};
			});
	}

	getDataWithUsers(items) {
		let requests = [];

		items.forEach(u => {
			requests.push(this.resolveUser(u.Username));
		});

		return Promise.all(requests).then(results => {
			let newItems = [];

			for (let i = 0; i < results.length; i++) {
				newItems.push({ ...items[i], user: results[i] });
			}

			return newItems;
		});
	}

	getAppType(session) {
		if (
			/android|blackberry|iphone|ipod|mobile|webos/i.test(
				session.userAgent
			)
		) {
			return 'mobile';
		}

		return 'app';
	}

	async loadData() {
		try {
			const service = await getService();
			const sessionsCollection = service.getWorkspace(ANALYTICS);
			const link = getLink(sessionsCollection, SESSIONS_LINK);
			const sessions = await service.get(link + '?limit=' + LIMIT);

			const resolvedItems = await this.getDataWithUsers(
				sessions.Items.slice(0, PAGE_SIZE)
			);

			this.setState({
				loading: false,
				allItems: sessions.Items,
				items: resolvedItems.map(x => {
					return {
						...x,
						appType: this.getAppType(x),
					};
				}),
			});
		} catch (e) {
			this.setState({ loading: false, items: [], error: e });
		}
	}

	renderHeader() {
		return (
			<div className="header">
				<div className="title">{t('title')}</div>
			</div>
		);
	}

	renderItem = (item, index) => {
		return <Item key={item.SessionStartTime} item={item} />;
	};

	renderItems() {
		const { items, loading } = this.state;

		if (loading) {
			return <Loading.Mask />;
		} else if (items && items.length === 0) {
			return <div className="no-items">{t('noItems')}</div>;
		}

		return (
			<div className="items-container">
				<div className="items">
					{(items || []).map(this.renderItem)}
				</div>
			</div>
		);
	}

	render() {
		return (
			<div className="dashboard-list-widget recent-sessions">
				{this.renderHeader()}
				{this.renderItems()}
			</div>
		);
	}
}
