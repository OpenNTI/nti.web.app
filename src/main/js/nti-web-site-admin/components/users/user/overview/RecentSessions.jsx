import React from 'react';
import PropTypes from 'prop-types';
import {DateTime, Loading} from '@nti/web-commons';
import {scoped} from '@nti/lib-locale';

const LABELS = {
	title: 'Recent Sessions',
	noSessions: 'No recent sessions',
	today: 'Today'
};

const t = scoped('nti-web-site-admins.components.users.user.overview.recentsessions', LABELS);

export default class RecentSessions extends React.Component {
	static propTypes = {
		historicalSessions: PropTypes.arrayOf(PropTypes.object),
		loading: PropTypes.bool
	}

	constructor (props) {
		super(props);
	}

	renderHeader () {
		return (<div className="header"><div className="title">{t('title')}</div></div>);
	}

	renderItem = (item) => {
		const today = new Date();
		const itemDate = new Date(item.SessionStartTime * 1000);

		let dayLabel = DateTime.format(itemDate, 'dddd');

		if(DateTime.format(today, 'L') === DateTime.format(itemDate, 'L')) {
			dayLabel = t('today');
		}

		return (
			<div key={item.SessionStartTime} className="item">
				<div className="time">{DateTime.format(itemDate, 'lll')}</div>
				<div className="day">{dayLabel}</div>
			</div>
		);
	}

	renderItems () {
		const { loading, historicalSessions } = this.props;

		if(loading) {
			return <Loading.Mask/>;
		}

		if(!historicalSessions || historicalSessions.length === 0) {
			return (<div className="no-sessions">{t('noSessions')}</div>);
		}

		return (
			<div className="items">
				{historicalSessions.map(this.renderItem)}
			</div>
		);
	}

	render () {
		return (
			<div className="user-recent-sessions">
				{this.renderHeader()}
				{this.renderItems()}
			</div>
		);
	}
}
