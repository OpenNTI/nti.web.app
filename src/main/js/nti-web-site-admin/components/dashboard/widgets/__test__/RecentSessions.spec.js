import React from 'react';
import renderer from 'react-test-renderer';
import { TestUtils } from '@nti/web-client';

import RecentSessions from '../RecentSessions';

const { tearDownTestClient, setupTestClient } = TestUtils;

const getMockService = (numberOfUsers) => {
	return {
		Items: [
			{
				hasLink: () => true,
				getLink: () => 'mockLink'
			}
		],
		getWorkspace: () => {
			return {};
		},
		get: () => {
			let Items = [];

			const startTime = (new Date().getTime() - 1000 * 60 * 60 * 3) / 1000;

			for(let i = 0; i < numberOfUsers; i++) {
				const index = i + 1;

				Items.push({
					Username: 'user' + index,
					getCreatedTime: () => new Date('10/30/2018'),
					userAgent: i % 2 === 0 ? 'android' : 'chrome',
					SessionStartTime: startTime,
					SessionEndTime: i % 2 === 0 ? null : startTime + 60 * 5
				});
			}

			return Promise.resolve({
				Items,
				Total: Items.length
			});
		}
	};
};

const onBefore = (numberOfUsers) => {
	jest.useFakeTimers();
	setupTestClient(getMockService(numberOfUsers));
};

const onAfter = () => {
	tearDownTestClient();
};

const flushPromises = () => new Promise(resolve => setImmediate(resolve));

/* eslint-env jest */
describe('Site admin dashboard widget recent sessions (5 sessions)', () => {
	beforeEach(() => onBefore(5));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<RecentSessions/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin dashboard widget recent sessions (10 sessions)', () => {
	beforeEach(() => onBefore(10));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<RecentSessions/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});

describe('Site admin dashboard widget recent sessions (no sessions)', () => {
	beforeEach(() => onBefore(0));
	afterEach(onAfter);

	test('Basic render test', async () => {
		const cmp = renderer.create(<RecentSessions/>);

		jest.runAllTimers();
		await flushPromises();
		jest.runAllTimers();

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
