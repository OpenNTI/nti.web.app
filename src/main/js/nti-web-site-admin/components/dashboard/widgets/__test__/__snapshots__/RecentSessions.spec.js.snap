// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site admin dashboard widget recent sessions (5 sessions) Basic render test 1`] = `
<div
  className="dashboard-list-widget recent-sessions"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      Recent Sessions
    </div>
  </div>
  <div
    className="items-container"
  >
    <div
      className="items"
    >
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user1
            </div>
            <div
              className="duration"
            >
              Ongoing
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="mobile-icon"
          >
            <div
              className="top"
            />
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user2
            </div>
            <div
              className="duration"
            >
              5m 0s
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="app-icon"
          >
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user3
            </div>
            <div
              className="duration"
            >
              Ongoing
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="mobile-icon"
          >
            <div
              className="top"
            />
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user4
            </div>
            <div
              className="duration"
            >
              5m 0s
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="app-icon"
          >
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user5
            </div>
            <div
              className="duration"
            >
              Ongoing
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="mobile-icon"
          >
            <div
              className="top"
            />
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Site admin dashboard widget recent sessions (10 sessions) Basic render test 1`] = `
<div
  className="dashboard-list-widget recent-sessions"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      Recent Sessions
    </div>
  </div>
  <div
    className="items-container"
  >
    <div
      className="items"
    >
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user1
            </div>
            <div
              className="duration"
            >
              Ongoing
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="mobile-icon"
          >
            <div
              className="top"
            />
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user2
            </div>
            <div
              className="duration"
            >
              5m 0s
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="app-icon"
          >
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user3
            </div>
            <div
              className="duration"
            >
              Ongoing
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="mobile-icon"
          >
            <div
              className="top"
            />
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user4
            </div>
            <div
              className="duration"
            >
              5m 0s
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="app-icon"
          >
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user5
            </div>
            <div
              className="duration"
            >
              Ongoing
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="mobile-icon"
          >
            <div
              className="top"
            />
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
      <div
        className="item"
      >
        <img
          className="avatar avatar avatar-color-6 item-image"
          src="unresolved-user-avatar.png"
        />
        <div
          className="info"
        >
          <div
            className="main-info"
          >
            <div
              className="name"
            >
              user6
            </div>
            <div
              className="duration"
            >
              5m 0s
            </div>
          </div>
          <div
            className="start-time"
          >
            3 hours ago
          </div>
        </div>
        <div
          className="value"
        >
          <div
            className="app-icon"
          >
            <div
              className="screen"
            />
            <div
              className="base"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Site admin dashboard widget recent sessions (no sessions) Basic render test 1`] = `
<div
  className="dashboard-list-widget recent-sessions"
>
  <div
    className="header"
  >
    <div
      className="title"
    >
      Recent Sessions
    </div>
  </div>
  <div
    className="no-items"
  >
    No sessions found
  </div>
</div>
`;
